flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/service/application/ScriptRunnerITCase.java

```diff
@@ -0,0 +1,274 @@
+        classNameCodes.put(
+                GENERATED_UPPER_UDF_CLASS,
+                String.format(GENERATED_UPPER_UDF_CODE, GENERATED_UPPER_UDF_CLASS));
+
+        udfJar =
+                UserClassLoaderJarTestUtils.createJarFile(
+                        Files.createTempDirectory(functionHome, "test-jar").toFile(),
+                        "test-classloader-udf.jar",
+                        classNameCodes);
+    }
+
+    @BeforeEach
+    void beforeEach() {
+        outputStream = new ByteArrayOutputStream(1024);
+        SqlDriver.enableTestMode(outputStream);
+    }
+
+    @AfterEach
+    void afterEach() throws Exception {
+        outputStream.close();
+        SqlDriver.disableTestMode();
+    }
+
+    @AfterAll
+    static void afterAll() {
+        org.apache.flink.core.testutils.CommonTestUtils.setEnv(originalEnv);
+    }
+
+    @Test
+    void testRunScriptFromFile(@TempDir Path workDir) throws Exception {
+        String script =
+                String.format(
+                        "CREATE TEMPORARY TABLE sink(\n"
+                                + "  a STRING\n"
+                                + ") WITH (\n"
+                                + "  'connector' = 'values'\n"
+                                + ");\n"
+                                + "ADD JAR '%s';\n"
+                                + "CREATE TEMPORARY FUNCTION lower_func AS '%s';\n"
+                                + "CREATE TEMPORARY VIEW v(c) AS VALUES ('A'), ('B'), ('C');\n"
+                                + "INSERT INTO sink SELECT lower_func(c) FROM v;",
+                        udfJar.getAbsolutePath(), GENERATED_LOWER_UDF_CLASS);
+
+        List<String> arguments =
+                Arrays.asList("--scriptPath", createStatementFile(workDir, script).toString());
+        runScriptInCluster(arguments);
+
+        assertThat(TestValuesTableFactory.getResultsAsStrings("sink"))
+                .containsExactly("+I[a]", "+I[b]", "+I[c]");
+    }
+
+    @Test
+    void testRunScriptFromRemoteFile(@TempDir Path workDir) throws Exception {
+        String script =
+                String.format(
+                        "CREATE TEMPORARY TABLE sink(\n"
+                                + "  a STRING\n"
+                                + ") WITH (\n"
+                                + "  'connector' = 'values'\n"
+                                + ");\n"
+                                + "ADD JAR '%s';\n"
+                                + "CREATE TEMPORARY FUNCTION lower_func AS '%s';\n"
+                                + "CREATE TEMPORARY VIEW v(c) AS VALUES ('A'), ('B'), ('C');\n"
+                                + "INSERT INTO sink SELECT lower_func(c) FROM v;",
+                        udfJar.getAbsolutePath(), GENERATED_LOWER_UDF_CLASS);
+        File file = createStatementFile(workDir, script).toFile();
+
+        try (MockHttpServer server = MockHttpServer.startHttpServer()) {
```

Comment:
That’s awesome! Using a mock HTTP server to test remote files is a great approach.